libcontainer/nsenter/nsexec.c

```diff
@@ -1225,8 +1233,24 @@ void nsexec(void)
 			 * was broken, so we'll just do it the long way anyway.
 			 */
 			write_log(DEBUG, "unshare remaining namespace (except cgroupns)");
-			if (unshare(config.cloneflags & ~CLONE_NEWCGROUP) < 0)
-				bail("failed to unshare remaining namespaces (except cgroupns)");
+			/*
+			 * In centos 7.4/7.5/7.6, when other processes read the /proc/pid/status
```

Comment:
Is this something that only affect old centos or is it something that still happens on modern (non-centos) kernels? I guess the latter?

If it is the former, I'd clarify this is a workaround for those kernels, if it is the latter I'd not tie the comment so much to these specific centos versions

Reply:
It's not just CentOS that has this bug, it's a kernel bug that was fixed in https://github.com/torvalds/linux/commit/12c641ab8270f787dfcce08b5f20ce8b65008096. The issue exists in older kernels, such as the v3.10.